Sources/MongoSwift/MongoCollection+BulkWrite.swift

```diff
@@ -412,38 +412,67 @@ public struct BulkWriteResult {
     }
 }
 
-/// A struct to represent a write error resulting from an executed bulk write.
-public struct WriteError: Codable {
-    /// The index of the request that errored.
-    public let index: Int
-
+/// A struct to represent a single write error not resulting from an executed bulk write.
+public struct WriteError: Codable, Equatable {
     /// An integer value identifying the error.
     public let code: Int
 
     /// A description of the error.
     public let message: String
 
     private enum CodingKeys: String, CodingKey {
-        case index
         case code
         case message = "errmsg"
     }
+
+    public static func == (lhs: WriteError, rhs: WriteError) -> Bool {
+        return lhs.code == rhs.code
+    }
 }
 
 /// A struct to represent a write concern error resulting from an executed bulk write.
-public struct WriteConcernError: Codable {
+public struct WriteConcernError: Codable, Equatable {
     /// An integer value identifying the write concern error.
     public let code: Int
 
     /// A document identifying the write concern setting related to the error.
-    public let info: Document
+    public let details: Document
+
+    /// A description of the error.
+    public let message: String
+
+    private enum CodingKeys: String, CodingKey {
+        case code
+        case details = "errInfo"
+        case message = "errmsg"
+    }
 
-    ///  A description of the error.
+    public static func == (lhs: WriteConcernError, rhs: WriteConcernError) -> Bool {
+        return lhs.code == rhs.code && lhs.details == rhs.details
+    }
+}
+
+/// A struct to represent a write error resulting from an executed bulk write.
+public struct BulkWriteError: Codable, Equatable {
+    /// An integer value identifying the error.
+    public let code: Int
```

Comment:
`code` here (and in other existing error usages) refers to libmongoc error code, right? 

while the upkeep may might be tedious, we should consider whether it's worth replicating all the libmongoc error codes in the driver as an enum with raw type `Int`, so that users don't have to go look at libmongoc docs to figure out what the codes corresponds to. 
alternatively, if we don't do that, we need to at least point users to the documentation for libmongoc error codes. I'm not sure what the best way to do that would be. it seems maybe less than desirable to just copy-paste the link every place we use an error code in a struct or enum case. we might consider introducing `public typealias ErrorCode = Int`, using that throughout the codebase, and just linking to it at the location where we define the `typealias`.
what do PHP and C++ do with respect to libmongoc error codes? 

regardless of what direction we go on that, I think it would be great if we add a ticket to the epic for writing a guide for our docs on error handling. (you can see we currently have one guide page [ in the docs](https://mongodb.github.io/mongo-swift-driver/development.html) already, the development guide.) this can explain the overall hierarchy and the types of errors to look out for, link to relevant pages/docs, and show some examples of best practices for catching and handling the errors.